Bump to latest upstream v1.x

lowdesertpunk opened the pull request.

eli-b committed:
Bundle fmt 9.1.0's std.h, and provide a header to include either it or the external fmt's version (#2539)

SpriteOvO committed:
Fix undefined macro `FMT_STRING` in benchmark when using `std::format` (#2540)

kromain committed:
Update #include to deprecated fmt header (#2545)

The <fmt/locale.h> header has been marked as deprecated for a while
and has finally been removed in fmt v0.9.0:
https://github.com/fmtlib/fmt/commit/5c7d315ded7bdb6cc5bd65daef091eefe

Replace with <fmt/format.h> instead, as recommended.

sylveon committed:
Support compile-time format string checking with std::format (#2544)

* Support compile-time format string checking with std::format

* Fix pre-VS 17.5 compilation

* Fix compilation without wchar_t support

* What am I doing

* Bring back fmt optimization

* Move to_string_view to common.h

* Fix SPDLOG_CONSTEXPR_FUNC emitting duplicate symbol errors when building in C++11

* Also add inline on VS 2013

* Appender doesn't work on wide strings

zydxhs committed:
fix dup_filter_sink lose source_loc (#2549)

Co-authored-by: zhuyadong <zhuyadong@kedacom.com>

zydxhs committed:
dup_filter_sink adds parameters to enable setting the level of skipped logs (#2563)

* dup_filter_sink adds parameters to enable setting the level of skipped logs

* rename the param name 'level' to 'notification_level'

Co-authored-by: zhuyadong <zhuyadong@kedacom.com>

koldi committed:
Unnecessary backtrace begin/end logs (#2568)

* add empty getter function to tracer

* add unit test to check empty tracer

Co-authored-by: Gyorgy Katona <gykatona@logmein.com>

gabime committed:
Added compile mscv_sink.h to tests

naddu77 committed:
Fixed variable name (#2573)

* fixed variable name

* Changed the variable name from check_debbugger_present_ to check_debugger_present_.

Co-authored-by: cpp <c.pp@navercorp.com>

Commit:
Feature/add system includes option (#2575)

* Adding system includes option

* Adding system includes option

alokpr committed:
feat(mpmc_blocking_q): add blocking dequeue without timeout (#2588)

Use the new blocking dequeue to avoid unnecessarily waking up the
thread pool every 10s.

Fixes #2587 by replacing std::condition_variable::wait_for with
std::condition_variable::wait as a workaroung for gcc 11.3 issue 101978.

Co-authored-by: Alok Priyadarshi <alokp@dexterity.ai>

segoon committed:
fix build: fix for freebsd (#2590)

The build error was:
  include/spdlog/details/tcp_client.h:106:31: error: use of undeclared identifier 'IPPROTO_TCP'

kraj committed:
Do not use LFS64 functions on linux/musl (#2589)

On musl, off_t is 64bit always ( even on 32bit platforms ), therefore
using LFS64 funcitons is not needed on such platforms. Moreover, musl
has stopped providing aliases for these functions [1] which means it
wont compile on newer musl systems. Therefore only use it on 32bit
glibc/linux platforms and exclude musl like cygwin or OSX

[1] https://git.musl-libc.org/cgit/musl/commit/?id=246f1c811448f37a44b41cd8df8d0ef9736d95f4
Signed-off-by: Khem Raj <raj.khem@gmail.com>

igrr committed:
Support newlib C library configurations without tm_gmtoff field (#2600)

Newlib C library (https://sourceware.org/newlib/) has a configuration
option to add tm_gmtoff field to the tm structure. Not all the
platforms supported by newlib enable this option, and spdlog doesn't
compile on such platforms due to missing tm_gmtoff field.

Fix this by checking for `__NEWLIB__` and `__TM_GMTOFF` and enabling
calculate_gmt_offset.

Tagl committed:
Fix type of event id in win_eventlog_sink (#2598)

Co-authored-by: Arnar Bjarni Arnarson <arnar@menandmice.com>

robinlinden committed:
Fix -Wshadow warnings in spdlog::sinks::dist_sink (#2599)

This is similar to fbba6dff20b0c04a0694515168914a62161999d7 but fixes a
few member functions missed in that commit.

gabime committed:
Update README.md

gabime committed:
Update README.md

albert-github committed:
Spelling corrections (#2606)

Spelling corrections v1.x

espkk committed:
Add sync to file_helper (#2343)

gabime committed:
Update os.h

gabime committed:
Update os-inl.h

gabime committed:
Update os.h

maghorbani committed:
Add a trivial callback sink (#2610)

Add a trivial callback sink

gabime committed:
Update README.md

gabime committed:
Update README.md

starsunyzl committed:
Fix unexpected delimiter at start of line in to_hex formatter (#2627)

zEuS0390 committed:
Fix MinGW build issue on example (#2642)

* Fix MinGW build issue on example #2638

* Move the cmake change to example\CMakeLists.txt

* Update CMakeLists.txt on the example

chardin-cpi committed:
Add optional TID definition to the systemd sink send (#2619)

From the systemd.journal-fields the TID is a user defined
field passed directly from the clients and stored in the
journal. Adding the arguement in the journal send to support
that storage option in the journal.

afshinpir committed:
Added `apply_logger_env_levels` (#2649)

This method applies levels which is set by environment variable
`SPDLOG_LEVEL` to the a single controller. Usefull for loading
configuration into manually created loggers.

Commit:
msvc_sink: support utf8 (#2651)

* msvc_sink: support utf8

gabime committed:
Better handling of utf to wchar

gabime committed:
move include cassert

gabime committed:
Update os-inl.h

gabime committed:
Update test_stopwatch.cpp

gabime committed:
Removed use of  SPDLOG_FMT_RUNTIME from test_errors.h

gabime committed:
Revert commit 0e9ccd73ef21015bff50c5044a88f94f33aff01b

gabime committed:
Update ci.yml

gabime committed:
Update ci.yml

gabime committed:
Update ci.yml

xvitaly committed:
Added Catch v3 support. (#2656)

gabime committed:
Update ci.yml

gabime committed:
revert pr #2656

xvitaly committed:
Allow other builders running after build failures. (#2659)

xvitaly committed:
Added Catch v3 support (#2661)

* Added Catch v3 support.

* Removed extra square brackets from some tests.

xvitaly committed:
Copy all compiled DLLs to correct destinations. (#2662)

gabime committed:
Update appveyor.yml

vbirds committed:
feat(kafka_skin.h): kafka log support (#2655)

* feat(kafka_skin.h): kafka log support

add kafka log support

* refactor(kafka_skin.h): remove producer_  check

remove producer_  check

gabime committed:
Update and rename kafka_skin.h to kafka_sink.h

gabime committed:
Update test_errors.cpp

gabime committed:
Update test_errors.cpp

gabime committed:
Update test_errors.cpp

MyNameIsLame committed:
Fix os namespace in systemd_sink.h (#2668)

* Fix os namespace in systemd_sink.h

* Remove spdlog:: prefix from os::thread_id() call

MyNameIsLame committed:
Enable systemd_sink tests in linux pipeline (#2669)

* Install libsystemd-dev in linux pipeline

Without this package the test_systemd_sink.cpp will not be tested.

* Install pkg-config in linux pipeline

bachittle committed:
setting the cmake standard to 20 when using std format (#2680)

scc-tw committed:
Fix stdout_sink_base::log's behavior inconsistency (#2646)

* Fix stdout_sink_base::log's behavior inconsistency

It will flush every time when it if not defined _WIN32, but not in
Windows family.
We viewed the commit #48d4ed9 for fixing issue #1675 .
It seems missing this flushing operation in mistake.

* Use fflush at all operating system

* Remove redundant fflush from stdout_sink_base

---------

Co-authored-by: scc <scc@teamt5.org>

bachittle committed:
fix small issue when compiling with C++20 without std::format (#2688)

H1X4Dev committed:
fix build for master fmt (non-bundled) (#2694)

* fix build for master fmt (non-bundled)

* update fmt_runtime_string macro

* fix build of updated macro

gabime committed:
Update registry-inl.h fix #2691

barracuda156 committed:
os-inl.h: fix for missing pthread_threadid_np (#2715)

jamesruan committed:
fix ringbuffer_sink moving warning (#2722)

zencatalyst committed:
Update README.md (#2732)

fixed serious grammar and spelling issues throughout the file without touching the content

gabime committed:
Update README.md

commel committed:
Removes special format handling for fmt.  (#2736)

* Removes special format handling for fmt. Regains test compatibility with fmt
1.10.0.

fixes #2735

* reverted std::vector back to filename_t and used pointer to array start likewise as fmt's implementation uses

* calc_filename buffer increase softened, exception is throw if buffer exceeds 4k, filename parameter renamed to match intend.

* calc_filetime based on std::put_time for simpler implementation

gabime committed:
Update daily_file_sink.h

gabime committed:
Update daily_file_sink.h

gabime committed:
Update daily_file_sink.h

gabime committed:
Added global logger benchmarks

gabime committed:
Catch exceptions from async logger. Fix #2618

gabime committed:
Update error message in default error handler

eli-b committed:
Fix fmt build (#2744)

gabime committed:
Update test_stopwatch.cpp

gabime committed:
Update test_stopwatch.cpp

gabime committed:
If exceptions are disabled, disable them in the bundled fmt as well

gabime committed:
Remov foreward to standard vformat_to

Ulmo-F committed:
qt_sink: add some warning on its usage (#2753)

* qt_sink: add some warning on its usage

* qt_sink: add some warning on its usage - fix

---------

Co-authored-by: Benoit FANCHON <bfanchon@nanoxplore.com>

gabime committed:
Update qt_sinks.h

gabime committed:
Update qt_sinks.h

gabime committed:
Update qt_sinks.h

gabime committed:
Fixed qt_sink

gabime committed:
Store MetaMethod object in qt_sink for better performance

gabime committed:
Clean qt_sink code

gabime committed:
Clean qt_sink code

gabime committed:
Added some comments to qt_sinks.h

gabime committed:
Added qt_color_sink

gabime committed:
Added default color handling to qt_color_sink

gabime committed:
Update comment about qt_color_sink

gabime committed:
Remove comment in qt_sinks.h

gabime committed:
Use at() in ansicolor_sink

gabime committed:
clang format

gabime committed:
Updated README.md with qt color example

gabime committed:
Update README.md

gabime committed:
Added lock to qt_color_sink

gabime committed:
Update qt_sinks.h

gabime committed:
Update qt_sinks.h

gabime committed:
Update qt_sinks.h

gabime committed:
Revert qt_sinks changes and color support, since they are not thread safe

gabime committed:
Update README.md

gabime committed:
wip color_qt_sink

gabime committed:
Cleaned qt_color_sink

gabime committed:
Updated README.md with qt color example

gabime committed:
Make max_line explicit for qt_color_sink

gabime committed:
Make max_line explicit for qt_color_sink

gabime committed:
Clean qt_color_sink

gabime committed:
Cleaned some warnings in qt_sinks

gabime committed:
Fixed clang warning in qt_sinks.h

7starsea committed:
Update INSTALL (#2775)

dayfixer committed:
Update qt_sinks.h: narrow cast msg.color_range_start, msg.color_range_end (#2781)

CChuancey committed:
ignore vscode and clangd cache files (#2787)

Co-authored-by: chuancey <chuancey@mail.com>

Simon-Janos committed:
Removing IPv4 limitation from tcp_client (#2790)

gabime committed:
Updated bundled fmt to version 10.0.0

gabime committed:
Added missing include to udp_client.h

gabime committed:
Fixed FMT_EXPORT to FMT_LIB_EXPORT in CMakeLists.txt

gabime committed:
Added const qualifier to stopwatch formatter to support c++20

gabime committed:
Added const qualifier to bin_hex_formatter to support c++20

gabime committed:
Added const to put_newline() in bin_to_hex.h

lucasrangit committed:
sinks: android: handle when message is not loggable (#2801)

Android logger (since API 30) checks the per-tag property `log.tag.<tag>` to determine if a log message is loggable. See https://developer.android.com/ndk/reference/group/logging#__android_log_is_loggable . For example, `__android_log_buf_write` for a VERBOSE message will call `__android_log_is_loggable` and return `-EPERM` if the log message will not be printed because `log.tag.<tag>` is set to `INFO`.

Instead of erroring with the following error message, the Android sink should handle `-EPERM`. It is not an error to disable a log via the run-time property.

```
[*** LOG ERROR #0001 ***] [2023-06-29 00:50:26] [logcat] logging to Android failed: Unknown error -1 [/path/to/file.cpp(123)]
```

gabime committed:
Update README.md

gabime committed:
Revert "Updated bundled fmt to version 10.0.0"

This reverts commit 62e55e7a7fb2fd3fd43fced10cccf166fcd8ff40.

gabime committed:
Revert "Fixed FMT_EXPORT to FMT_LIB_EXPORT in CMakeLists.txt"

This reverts commit c3fa8f60e2aa6fd6257016436794c024c360cec8.

gabime committed:
Bumped spdlog version to 1.12.0

gabime committed:
Formatted qt_sinks.h code

gabime committed:
Fixed #2724 by excluding bin_to_hex sink if using std::format

krawq committed:
Qt Sink : Allow for darker colors (for light background). (#2817)

Default are too bright if background is light(white).

lowdesertpunk committed:
Merge upstream/v1.x with conflicts

lowdesertpunk committed:
Resolve conflicts

gabime committed:
Fix #2820

xvitaly committed:
Added missing square bracket to fix the level_to_string_view test. (#2827)

moritz-h committed:
Set CMAKE_BUILD_TYPE only for single-config generators (#2839)

Thanks @moritz-h

gabime committed:
Fixed circular_q size impl and added tests

gabime committed:
Fixed ci

gabime committed:
Added another test for circular_q

robertmaynard committed:
Provide spdlog_header_only in build directory export (#2846)

Fixes #2678

mentlerd commented on .github/workflows/ci.yml:
Are we not concerned about 3rd party forks running CI actions within the org? We should probably disable GH actions on these repos.

lowdesertpunk replied:
Why? Even though they fail right now, could indicate proper functionality.

mentlerd commented on include/spdlog/details/os-inl.h:
```diff
@@ -353,7 +360,22 @@ SPDLOG_INLINE size_t _thread_id() SPDLOG_NOEXCEPT
     uint64_t tid;
+    // There is no pthread_threadid_np prior to 10.6, and it is not supported on any PPC,
+    // including 10.6.8 Rosetta. __POWERPC__ is Apple-specific define encompassing ppc and ppc64.
+#    if (MAC_OS_X_VERSION_MAX_ALLOWED < 1060) || defined(__POWERPC__)
```
Is this going to work with our iOS builds?

lowdesertpunk replied:
Undefined identifiers are evaluated to 0. Let me check.

lowdesertpunk replied:
It works, but the runtime check is unnecessary. https://github.com/gabime/spdlog/pull/2897 is already in upstream master, updating this PR to latest master.

lowdesertpunk replied:
Updated to latest upstream master. No other notable changes since the previous update.

mentlerd commented on include/spdlog/spdlog.h:
```diff
@@ -131,50 +131,59 @@ SPDLOG_API spdlog::logger *default_logger_raw();
+//
+// Example:
+//   auto mylogger = std::make_shared<spdlog::logger>("mylogger", ...);
+//   spdlog::apply_logger_env_levels(mylogger);
```
Looks like we missed this and implemented our own. Perhaps we should/could replace our internal solution?

lowdesertpunk replied:
I guess missed because it's wasn't available at that time. We can replace ours, but ideally we should switch to some configuration file based solution to allow settings up sinks, etc. as well without having to recompile.

kegechen committed:
Add SPDLOG_TO_VERSION to compare spdlog version (#2853)

You can use SPDLOG_VERSION to select the latest spdlog features
where available while falling back to older implementations otherwise.
Using SPDLOG_TO_VERSION() for the value to compare with is recommended.
for Example:
```c++
 void sink_it_(const details::log_msg &msg) override
 {
 #if SPDLOG_VERSION < SPDLOG_TO_VERSION(1,4,0)
     fmt::memory_buffer formatted;
 #else
     memory_buf_t formatted;
 #endif
     sinks::base_sink<Mutex>::formatter_->format(msg, formatted);
     // bala bala...
 }
```

albert-github committed:
Remove obsolete part from cmake configuration files (#2871)

Updating minimum CMake version to 3.11

gabime committed:
Update ci.yml

gabime committed:
Removed policy_max from  cmake_minimum_required(..)

gabime committed:
Fixed bench

neothenil committed:
Fix encoding issue in qt_sinks (#2862)

Added support for utf8 in qt_color_sink

gabime committed:
clang format

gabime committed:
minor cleanup

walkerlala committed:
Support async_overflow_policy::discard_new (#2876)

Reason for the discard_new policy: when there is an overflow, there
is usually some unexpected issue (a bug, or some other unexpected stuff).
And in case of unexpected issue, the first arrived log messages are usually
more important than subsequent ones. For example, some application
keep logging error messages in case of functionality failure, which,
when using async_overflow_policy::overrun_oldest, will overrun the
first arrived messages that may contain real reason for the failure.

gabime committed:
Fix code formatting of async_logger-inl.h

gabime committed:
Update async_logger-inl.h

gabime committed:
Updated format.sh script

gabime committed:
Updated clang format to google style

gabime committed:
Added missing include mutex

gabime committed:
clang format

gabime committed:
Update common.h

gabime committed:
Fixed windows compile

gabime committed:
Never sort includes in clang format

gabime committed:
format

gabime committed:
updated .clang-format

gabime committed:
updated format.sh

gabime committed:
clang format

gabime committed:
updated clang format

gabime committed:
reformat code

gabime committed:
update clang format again

gabime committed:
update clang format again

oddko committed:
Added a function to add callbacks that are called when a logger is registered (#2883)

* Added a function to add callbacks that are called when a logger is registered

* Fix non captured registration 2 not being properly tested for

* Replace std::list by std::vector

* Remove const refs to shared pointers

* Fix missing header

gabime committed:
Revert "Added a function to add callbacks that are called when a logger is registered (#2883)"

This reverts commit b6eeb7364cb2f6eeb1ce963d0fa6ddbc929861f4, since it causes deadlocks too easily for the users.

gabime committed:
Update README.md

gabime committed:
Update README.md

lowdesertpunk committed:
Fix OS availability check of pthread_threadid_np for iOS (#2897)

lowdesertpunk committed:
Merge upstream/v1.x with conflicts

lowdesertpunk committed:
Add .git-blame-ignore-revs to ignore clang-format related commits (#2899)

lowdesertpunk committed:
Resolve conflicts

lowdesertpunk committed:
Merge upstream/v1.x
